frontend/src/components/card/UserCard/UserCard.tsx

```diff
@@ -0,0 +1,96 @@
+/*
+
+user card component
+
+how to use: <UserCard userInfo={userInfo} cardType={cardType} />
+
+usage tip:
+1. locate this component in to the container which has flex flex-col px-1 style
+
+props for UserCard: { 
+  1. userInfo: {id, name, profile_image, current_status, introduction} 
+  2. cardType: {basic, friend, noneFriend, friendRequest, blockedUser}
+}
+
+buttons for each cardType:
+1. basic: none
+2. friend: match, unfollow, block
+3. noneFriend: follow, block
+4. friendRequest: follow, unfollow, block
+5. blockedUser: unblock
+
+NOTE: There is a unsolved bug at the sub component UserCardButton.tsx. 
+but it will be fixed soon. 
+
+TODO: fix the bug at UserCardButton.tsx
+
+*/
+
+import UserCardButton from './UserCardButton';
+import UserCardAvatar from './UserCardAvatar';
+
+import {Card, CardDescription, CardTitle} from '@/components/shadcn/card';
+
+export default function UserCard(props: any) {
```

Comment:
프롭스 타입 지정이 any인 이유가 잇나욤?!

Reply:
뭐로 해야 할지 몰라서 ㅠㅠ 혹시 더 좋은 방법 있을까요

Reply:
형이 위에 프롭스 잘 정리해줫어가지고 저거 인터페이스로 정의해서 넣어주심 타입스크립트의 장점을 살리면서 쓸 수 있을 것 같아욤
```suggestion
export default function UserCard(props: any) {
```